firedrake/mesh.py

```diff
@@ -1063,15 +1081,25 @@ def _facets(self, kind):
                        markers=base.markers,
                        unique_markers=base.unique_markers)
 
-    def make_cell_node_list(self, global_numbering, entity_dofs, offsets):
+    def make_cell_node_list(self, global_numbering, entity_dofs, entity_permutations, offsets):
         """Builds the DoF mapping.
 
         :arg global_numbering: Section describing the global DoF numbering
         :arg entity_dofs: FInAT element entity DoFs
+        :arg entity_permutations: FInAT element entity permutations
         :arg offsets: layer offsets for each entity dof.
         """
+        if entity_permutations is None:
+            # FInAT entity_permutations not yet implemented
+            entity_dofs = eutils.flat_entity_dofs(entity_dofs)
+            return super().make_cell_node_list(global_numbering, entity_dofs, None, offsets)
+        assert sorted(entity_dofs.keys()) == sorted(entity_permutations.keys()), "Mismatching dimension tuples"
+        for key in entity_dofs.keys():
+            assert sorted(entity_dofs[key].keys()) == sorted(entity_permutations[key].keys()), "Mismatching entity tuples"
+        assert all(v == 0 or v == 1 for b, v in entity_permutations), "Vertical dim index must be in [0, 1]"
```

Comment:
Style: `all(v in {0, 1} for _, v in entity_permutations)`

Reply:
Thanks. Fixed.